### PR TITLE
refactor(fs): cleanup use of `@std/assert`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #4865

### CODE REVIEW

File: fs/copy.ts [hunk 1]
```diff
@@ -34,6 +33,12 @@ interface InternalCopyOptions extends CopyOptions {
   isFolder?: boolean;
 }
 
+function assertIsDate(date: Date | null, name: string): asserts date is Date {
+  if (date === null) {
+    throw new ReferenceError(`${name} is unavailable`);
```
Comment (kt3k) [inline fs/copy.ts:38]: This usage of `ReferenceError` seems strange to me (It's not about variable being unavailable). I think this should be `TypeError` or `Error`.

File: fs/expand_glob.ts [hunk 2]
```diff
@@ -158,7 +157,9 @@ export async function* expandGlob(
     : absRoot;
   while (segments.length > 0 && !isGlob(segments[0]!)) {
     const seg = segments.shift();
-    assert(seg !== undefined);
+    if (seg === undefined) {
+      throw new ReferenceError("Unexpected undefined segment");
```
Comment (kt3k) [inline fs/expand_glob.ts:161]: Ditto. I think this should be `TypeError` or `Error`

File: fs/expand_glob.ts [hunk 3]
```diff
@@ -316,7 +317,9 @@ export function* expandGlobSync(
...
```
Comment (kt3k) [inline fs/expand_glob.ts:321]: Ditto. I think this should be `TypeError` or `Error`

File: fs/copy.ts (hunk 1 above)
Comment (iuioiua) [reply fs/copy.ts:38]: I was going to use `TypeError` originally. But we pass [`Deno.FileInfo.atime`](https://deno.land/api?s=Deno.FileInfo#prop_atime) and [`Deno.FileInfo.mtime`](https://deno.land/api?s=Deno.FileInfo#prop_mtime) to this function, which can only be `null` on some platforms (see documentation). So I went with `ReferenceError`. WDYT?

Comment (kt3k) [reply fs/copy.ts:38]: `ReferenceError` is thrown for the case where undeclared 'variable' used like the below:

```ts
const foo = a; // a is not declared
```

There seems no undeclared variable involved in this case. So I think this case doesn't fit to `ReferenceError`

Comment (iuioiua) [reply fs/copy.ts:38]: Ah, yeah, fair point. I'll go with `Error`.

File: fs/expand_glob.ts (hunk 2 above)
Comment (iuioiua) [reply fs/expand_glob.ts]: I went with `TypeError`.

File: fs/copy.ts (hunk 1 above)
Comment (iuioiua) [reply fs/copy.ts]: I went with `Error`.

**Review by kt3k — APPROVED** [review]
LGTM. Nice!

**Pull request merged**